src/commands/register.ts

```diff
@@ -118,40 +127,61 @@ export default class ComponentRegister extends Command {
     const project_name = `register.${resourceRefToNodeRef(component_spec.name)}.${tag}`;
     const compose_file = DockerComposeUtils.buildComposeFilepath(this.app.config.getConfigDir(), project_name);
 
-    await DockerComposeUtils.writeCompose(compose_file, yaml.dump(compose));
+    await DockerBuildXUtils.writeCompose(compose_file, yaml.dump(compose));
 
     let build_args: string[] = [];
     for (const service_config of Object.values(component_spec.services || {})) {
       build_args = build_args.concat((await this.getBuildArgs(service_config)).map(arg => {
         return `${arg}`;
       }));
     }
+
     build_args = build_args.filter((value, index, self) => {
       return self.indexOf(value) === index;
     }).reduce((arr, value) => {
-      arr.push('--build-arg');
-      arr.push(value);
+      arr.push("--set");
+      arr.push(`*.args.${value}`);
       return arr;
     }, [] as string[]);
 
     try {
-      await DockerComposeUtils.dockerCompose(['-f', compose_file, 'build', ...build_args], {
-        stdio: 'inherit',
+      const is_local = this.app.config.api_host.includes("localhost");
+      if (is_local) {
+        // Create a configuration file for buildkitd
+        const local_buildkitd_config_file = "buildkitd.toml";
+        const file_content = `[registry.\"${this.app.config.registry_host}\"]\n  http = true\n  insecure = true`;
+        await DockerBuildXUtils.writeBuildkitdConfigFile(local_buildkitd_config_file, file_content);
+    
+        await DockerBuildXUtils.dockerBuildX(["create", "--name", "architect", "--driver-opt", "network=host", "--use", "--buildkitd-flags", "--allow-insecure-entitlement security.insecure", `--config=${local_buildkitd_config_file}`], {
+          stdio: "inherit",
+        });
+      } else {
+        await DockerBuildXUtils.dockerBuildX(["create", "--name", "architect"], {
+          stdio: "inherit",
+        });
+      }
+    } catch (err: any) {
+      this.log(`Docker builder instance 'architect' already exists. Using existing 'architect' builder instance ...`);
+    }
+
+    try {
+      await DockerBuildXUtils.dockerBuildX(["inspect", "--bootstrap"], {
+        stdio: "inherit",
       });
     } catch (err: any) {
       fs.removeSync(compose_file);
-      this.log(`Docker build failed. If an image is not specified in your component spec, then a Dockerfile must be present`);
+      this.log(`Docker buildx failed to inspect`);
```

Comment:
Change this too
```
Unable to start buildx builder. Please make sure docker is running.
```
Inspect may be a confusing word if people don't have context into buildx.